src/main/java/com/github/swrirobotics/bags/BagService.java

```diff
@@ -580,6 +583,58 @@ public String getVehicleName(BagFile bag) {
         return null;
     }
 
+    public String getMetadata(BagFile bagFile) {
+        String[] topics = myConfigService.getConfiguration().getMetadataTopics();
+        try {
+            for (String topic : topics) {
+                com.github.swrirobotics.bags.reader.messages.serialization.MessageType mt = bagFile.getFirstMessageOnTopic(topic);
+                int index = 1;
+                try {
+                    // Advance to newest
+                    index++;
+                    mt = bagFile.getMessageOnTopicAtIndex(topic, index);
+                } catch (java.lang.IndexOutOfBoundsException e) {
+                    ;
+                }
+                if (mt != null) {
+                    return mt.<StringType>getField("data").getValue();
+                }
+            }
+        } catch (BagReaderException | UninitializedFieldException | java.util.NoSuchElementException e) {
+            reportStatus(Status.State.ERROR,
+                    "Unable to get metadata from bag file " + bagFile.getPath() + ": " + e.getLocalizedMessage());
+        }
+        return new String();
+    }
+
+    public Set<Tag> getTags(BagFile bagFile, Bag bag) {
+        String metadata = getMetadata(bagFile);
+        Set<Tag> tags = new HashSet<Tag>();
+        String[] lines = metadata.split(System.getProperty("line.separator"));
+        for (String line : lines) {
+            if (line.isEmpty() | !line.contains(":"))
+                continue;
```

Comment:
I think you probably meant `(line.isEmpty() || !line.contains(":"))` here; also, please put brackets around single-line conditionals.